x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/job/DateHistogramGroupConfig.java

```diff
@@ -47,71 +52,80 @@
  *     ]
  * }
  */
-public class DateHistoGroupConfig implements Writeable, ToXContentFragment {
-    private static final String NAME = "date_histo_group_config";
-    public static final ObjectParser<DateHistoGroupConfig.Builder, Void> PARSER
-            = new ObjectParser<>(NAME, DateHistoGroupConfig.Builder::new);
-
-    private static final ParseField INTERVAL = new ParseField("interval");
-    private static final ParseField DELAY = new ParseField("delay");
-    private static final ParseField FIELD = new ParseField("field");
-    public static final ParseField TIME_ZONE = new ParseField("time_zone");
+public class DateHistogramGroupConfig implements Writeable, ToXContentObject {
+
+    private static final String NAME = "date_histogram";
+    private static final String INTERVAL = "interval";
+    private static final String FIELD = "field";
+    public static final String TIME_ZONE = "time_zone";
+    private static final String DELAY = "delay";
+    private static final String DEFAULT_TIMEZONE = "UTC";
+    private static final ConstructingObjectParser<DateHistogramGroupConfig, Void> PARSER;
+    static {
+        PARSER = new ConstructingObjectParser<>(NAME, a ->
+            new DateHistogramGroupConfig((String) a[0], (DateHistogramInterval) a[1], (DateHistogramInterval) a[2], (String) a[3]));
+        PARSER.declareString(constructorArg(), new ParseField(FIELD));
+        PARSER.declareField(constructorArg(), p -> new DateHistogramInterval(p.text()), new ParseField(INTERVAL), ValueType.STRING);
+        PARSER.declareField(optionalConstructorArg(),  p -> new DateHistogramInterval(p.text()), new ParseField(DELAY), ValueType.STRING);
+        PARSER.declareString(constructorArg(), new ParseField(TIME_ZONE));
+    }
 
-    private final DateHistogramInterval interval;
     private final String field;
-    private final DateTimeZone timeZone;
+    private final DateHistogramInterval interval;
     private final DateHistogramInterval delay;
+    private final String timeZone;
+
+    public DateHistogramGroupConfig(final String field,
+                                    final DateHistogramInterval interval,
+                                    final @Nullable DateHistogramInterval delay,
+                                    final String timeZone) {
```

Comment:
Can you add javadocs to this constructor and maybe add the fact that we use Joda to parse the timeZone. Also the timeZone is optional so we could maybe have a simple ctr for `field` and `interval` and one that have all the options ?

Reply:
++

Also, could we annotate `timeZone` as `@Nullable` here too?